src-java/kilda-persistence-tinkerpop/src/test/java/org/openkilda/persistence/ferma/repositories/FermaFlowMirrorRepositoryTest.java

```diff
@@ -0,0 +1,171 @@
+/* Copyright 2022 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.ferma.repositories;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.openkilda.model.FlowMirror;
+import org.openkilda.model.FlowPathStatus;
+import org.openkilda.model.Switch;
+import org.openkilda.model.SwitchId;
+import org.openkilda.persistence.inmemory.InMemoryGraphBasedTest;
+import org.openkilda.persistence.repositories.FlowMirrorRepository;
+import org.openkilda.persistence.repositories.SwitchRepository;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class FermaFlowMirrorRepositoryTest extends InMemoryGraphBasedTest {
+    static final String TEST_FLOW_MIRROR_ID_1 = "flow_mirror_1";
+    static final String TEST_FLOW_MIRROR_ID_2 = "flow_mirror_2";
+    static final String TEST_FLOW_MIRROR_ID_3 = "flow_mirror_3";
+    static final SwitchId TEST_SWITCH_A_ID = new SwitchId(1);
+    static final SwitchId TEST_SWITCH_B_ID = new SwitchId(2);
+    static final SwitchId TEST_SWITCH_C_ID = new SwitchId(3);
+
+    FlowMirrorRepository flowMirrorRepository;
```

Comment:
All these members could be private or is there a reason not to do so?

Reply:
There is no reason. We can make it private